Model/ModelManager.php

```diff
@@ -376,7 +386,7 @@ public function getNormalizedIdentifier($entity)
         }
 
         // the entities is not managed
-        if (!$entity /*|| !$this->getEntityManager($entity)->getUnitOfWork()->isInIdentityMap($entity) // commented for perfomance concern */) {
+        if (!$entity || !$this->getEntityManager($entity)->getUnitOfWork()->isInIdentityMap($entity)) {
```

Comment:
(sorry, in the original code, it just does a void `return` ... I added the nulls because the 'success' case returns something and my IDE was complaining about an inconsitent return point)